bfd/elflink.c

```diff
@@ -13564,6 +13564,13 @@ _bfd_elf_section_already_linked (bfd *abfd,
 
   flags = sec->flags;
 
+  /* Check if the section type is SHT_COMDAT and the SEC_LINK_ONCE flag is not
```

Comment:
That may be solved using groups and comdat. From as manual:

.section name , "flags"G, @type, GroupName[, linkage]
The GroupName field specifies the name of the section group to which this particular
section belongs. The optional linkage field can contain:
- **comdat** indicates that only one copy of this section should be retained
- **.gnu.linkonce** an alias for comdat

Anyhow, we need to relay on gnu.linkonce as this is the default gnu way of handling this. A 100% metware compatibility may not be possible. Alternatively, we can ask metaware guys to support gnu like names.